src/batchview.jl

```diff
@@ -101,24 +112,32 @@ Return the fixed size of each batch in `data`.
 batchsize(A::BatchView) = A.batchsize
 
 Base.length(A::BatchView) = A.count
+
 getobs(A::BatchView) = getobs(A.data)
-getobs(A::BatchView, i::Int) = getobs(A.data, _batchrange(A, i))
 
-function getobs(A::BatchView, is::AbstractVector)
+function Base.getindex(A::BatchView, i::Int)
+    obsindices = _batchrange(A, i)
+    _getbatch(A, obsindices)
+end
+
+function Base.getindex(A::BatchView, is::AbstractVector)
     obsindices = union((_batchrange(A, i) for i in is)...)::Vector{Int}
+    _getbatch(A, obsindices)
+end
+
+function _getbatch(A::BatchView{TElem, TData, Val{true}}, obsindices) where {TElem, TData}
+    batch([getobs(A.data, i) for i in obsindices])
+end
+function _getbatch(A::BatchView{TElem, TData, Val{false}}, obsindices) where {TElem, TData}
+    return [getobs(A.data, i) for i in obsindices]
+end
+function _getbatch(A::BatchView{TElem, TData, Val{nothing}}, obsindices) where {TElem, TData}
     getobs(A.data, obsindices)
 end
 
 Base.parent(A::BatchView) = A.data
 Base.eltype(::BatchView{Tel}) where Tel = Tel
 
-Base.getindex(A::BatchView, i::Int) = obsview(A.data, _batchrange(A, i))
-
-function Base.getindex(A::BatchView, is::AbstractVector)
-    obsindices = union((_batchrange(A, i) for i in is)...)::Vector{Int}
-    obsview(A.data, obsindices)
-end
```

Comment:
So this PR is changing the indexing behavior from view to materialization.
I think this is the right direction, consistently with Base's array interface and with the fact that getobs falls back on indexing.

Reply:
> > getobs(A::BatchView) is currently defined as getobs(A.data) which doesn't do any batching. I would have thought that it should simply return all the batches
> 
> agree

I changed the method to return batches 👍